vignettes/fims-path-maturity.Rmd

```diff
@@ -0,0 +1,614 @@
+
+The vignette will build up components to the final path:
+```{r image-path-maturity-8a, echo = FALSE, message=FALSE, out.width = '100%'}
+knitr::include_graphics("figures/fims-path-maturity-8.png")
+```
+
+Code chunks in this vignette will be color coded similar to this diagram: R code in blue, Rcpp interface code in orange, fims core code in green, and information in grey.
+
+## Setting up a module in R
+
+We'll start from R. The first step is to install FIMS, retrieve the FIMS module, create a new maturity module inside FIMS, and populate the maturity module with parameter values. 
+
+```{r, class.source = "rchunk", eval = FALSE}
+# install and load FIMS
+remotes::install_github("NOAA-FIMS/FIMS@integration-test-estimation-R-clear-FIMS")
+library(FIMS)
+# Create a new maturity model
+fims <- Rcpp::Module("fims", PACKAGE = "FIMS")
+maturity <- new(fims$LogisticMaturity)
+# Populate the maturity module with parameter values.
+maturity$median$value <- 10
+maturity$median$is_random_effect <- FALSE
+maturity$median$estimated <- FALSE
+maturity$slope$value <- 0.2
+maturity$slope$is_random_effect <- FALSE
+maturity$slope$estimated <- FALSE
+```
+
+After that, we need to create and set up a population and link the maturity module to the population:
+```{r,  class.source = "rchunk", eval = FALSE}
+population <- new(fims$Population)
+#set up population specific parameters (other modules)
+#...
+#set maturity for the population - will throw error until PR #363 is merged into main
+population$SetMaturity(maturity$get_id())
+```
+
+Because FIMS sets up objects stored in memory, simply running `rm(fims)`, won't free this memory back up. There is a function that can be run to clear out memory:
+```{r,  class.source = "rchunk", eval = FALSE}
+# Clear C++ memory
+fims$clear()
+#Clear R memory
+rm(fims)
+rm(maturity)
+rm(population)
+```
+Also restarting or closing out your R session will free up memory.
+
+## The Rcpp Interface
+
+The fields specific to the maturity module that are accessible from R are defined in the 
+[rcpp_interface.hpp](https://github.com/NOAA-FIMS/FIMS/blob/main/inst/include/interface/rcpp/rcpp_interface.hpp) file under [inst/include/interface/rcpp](https://github.com/NOAA-FIMS/FIMS/tree/main/inst/include/interface/rcpp):
+
+```{Rcpp, class.source = "rcppchunk", eval = FALSE}
+Rcpp::class_<LogisticMaturityInterface>("LogisticMaturity")
+      .constructor()
+      .field("median", &LogisticMaturityInterface::median)
+      .field("slope", &LogisticMaturityInterface::slope)
+      .method("get_id", &LogisticMaturityInterface::get_id)
+      .method("evaluate", &LogisticMaturityInterface::evaluate);
+```
+
+The code above shows that there are two maturity module-specific fields, `median` and `slope`, and two maturity module-specific methods (i.e. functions), `get_id` and `evaluate`. You can access the fields and methods from R using the `show(fims)` function.
```

Comment:
quibble: All instances of "i.e." and "e.g." should be followed by a comma because we are using American English rather than British English. See [jakubmarian](https://jakubmarian.com/comma-after-i-e-and-e-g/) for a more detailed explanation.

suggestion: Add "(i.e., parameters)" after "fields" to match the 👍text👍 explaining "method".